File: sentry-android/src/main/java/com/joshdholtz/sentry/SentryEventRequest.java
package com.joshdholtz.sentry;

import org.json.JSONObject;

import java.io.Serializable;
import java.util.UUID;

/**
 * Created by Hazer on 2/25/16.
 */
class SentryEventRequest implements Serializable {
    private String requestData;
    UUID uuid;

    public SentryEventRequest(SentryEventBuilder builder) {
        this.requestData = new JSONObject(builder.getEvent()).toString();
        this.uuid = UUID.randomUUID();
    }

    /**
     * @return the requestData
     */
    public String getRequestData() {
        return requestData;
    }

    /**
     * @return the uuid
     */
    public UUID getUuid() {
        return uuid;
    }

    @Override
    public boolean equals(Object other) {
        SentryEventRequest otherRequest = (SentryEventRequest) other;

        if (this.uuid != null && otherRequest.uuid != null) {
            return uuid.equals(otherRequest.uuid);
        }

        return false;
    }

}

File: sentry-android/src/main/java/io/vithor/sentry/raven/SentryEventRequest.kt
package io.vithor.sentry.raven

import org.json.JSONObject

import java.io.Serializable
import java.util.UUID

/**
 * Created by Hazer on 2/25/16.
 */
internal class SentryEventRequest(builder: SentryEventBuilder) : Serializable {
    /**
     * @return the requestData
     */
    val requestData: String

    /**
     * @return the uuid
     */
    val uuid: UUID

    init {
        this.requestData = JSONObject(builder.event).toString()
        this.uuid = UUID.randomUUID()
    }

    override fun equals(other: Any?): Boolean {
        val otherRequest = other as SentryEventRequest?
        return uuid == otherRequest?.uuid
    }

    override fun hashCode(): Int{
        var result = requestData.hashCode()
        result += 31 * result + uuid.hashCode()
        return result
    }

}
